packages/core/src/utils/requestQueue.ts

```diff
@@ -33,6 +36,24 @@ export type RequestQueue = Omit<
   request: <TParameters extends EIP1193Parameters<PublicRpcSchema>>(
     parameters: TParameters,
   ) => Promise<RequestReturnType<TParameters["method"]>>;
+  subscribe: (params: SubscribeParameters) => Promise<SubscribeReturnType>;
+};
+
+export type SubscribeParameters = Parameters<
+  ResolvedWebSocketTransport["value"]["subscribe"]
+>[0] & {
+  method: "eth_subscribe";
+};
+
+export type SubscribeReturnType = Awaited<
+  ReturnType<ResolvedWebSocketTransport["value"]["subscribe"]>
+>;
```

Comment:
Are these types from viem?

Reply:
Yes, `ResolvedWebSocketTransport` is based on the `WebSocketTransport` type defined in viem. 
```ts
type ResolvedWebSocketTransport = Omit<
  ReturnType<WebSocketTransport>,
  "value"
> & {
  value: NonNullable<ReturnType<WebSocketTransport>["value"]>;
};
```